app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,48 @@
+require 'date'
+
+class SalesController < ApplicationController
+
+  def show
+    @sale = Sale.where(vendor_id: params[:id]).order(purchase_time: :asc)
+    @group = Sale.group("strftime('%m', purchase_time)").count
+    @current = Time.now.strftime('%m')
+    render :show
+  end
+
+  def month_sale
```

Comment:
Again a lot of logic happening in the controller - probably more appropriate as a method in the model.